kernel/main.c

```diff
@@ -340,6 +353,52 @@ async_init(void)
       panic("Unable to create a kthread for do_async_init()");
 }
 
+static void
+setup_uefi_runtime_services() {
+   EFI_RUNTIME_SERVICES *RT = TO_PTR(uefi_rt_addr);
+
+   printk("Runtime Services: %p\n", RT);
+
+   struct mem_region ma;
+
+   EFI_MEMORY_DESCRIPTOR *virt_map = kmalloc((size_t)get_mem_regions_count());
```

Comment:
It doesn't matter for the sake of the experiment, but for the final change don't waste memory: first count the number of regions with a dedicated loop, then allocate the map and then fill it.